<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>*brainClass*: Get started • brainClass</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="*brainClass*: Get started">
<meta property="og:description" content="brainClass">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">brainClass</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/brainClass.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/Mengbo-Li/brainClass/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="brainClass_files/header-attrs-2.8/header-attrs.js"></script><script src="brainClass_files/htmlwidgets-1.5.3/htmlwidgets.js"></script><script src="brainClass_files/jquery-3.5.1/jquery.min.js"></script><link href="brainClass_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="brainClass_files/datatables-binding-0.18/datatables.js"></script><link href="brainClass_files/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="brainClass_files/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="brainClass_files/dt-core-1.10.20/js/jquery.dataTables.min.js"></script><link href="brainClass_files/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet">
<script src="brainClass_files/crosstalk-1.1.1/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>
<em>brainClass</em>: Get started</h1>
                        <h4 class="author">Mengbo Li</h4>
            <address class="author_afil">
      School of Mathematics and Statistics, The University of Sydney, Australia<br><small class="dont-index">Source: <a href="https://github.com/Mengbo-Li/brainClass/blob/master/vignettes/brainClass.Rmd"><code>vignettes/brainClass.Rmd</code></a></small>
      <div class="hidden name"><code>brainClass.Rmd</code></div>

    </address>
</div>

    
    
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p>Here we present an example of the <em>brainClass</em> workflow to perform transcriptional-data-guided fMRI network classification and the post-hoc evaluation using the COBRE data set.</p>
</div>
<div id="install-brainclass" class="section level1">
<h1 class="hasAnchor">
<a href="#install-brainclass" class="anchor"></a>Install <code>brainClass</code>
</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install.packages('devtools')</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"Mengbo-Li/brainClass"</span><span class="op">)</span></code></pre></div>
</div>
<div id="load-packages" class="section level1">
<h1 class="hasAnchor">
<a href="#load-packages" class="anchor"></a>Load packages</h1>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Mengbo-Li/brainClass">brainClass</a></span><span class="op">)</span>
<span class="co">## if needed</span>
<span class="co">## devtools::install_github('jesusdaniel/graphclass')</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">graphclass</span><span class="op">)</span></code></pre></div>
</div>
<div id="load-data" class="section level1">
<h1 class="hasAnchor">
<a href="#load-data" class="anchor"></a>Load data</h1>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"ahba"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"gscv7.0"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"digsee"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"COBRE.data"</span><span class="op">)</span></code></pre></div>
<ul>
<li>Gene expression data from Allen Human Brain Atlas (AHBA)</li>
</ul>
<p>This is an ROI-by-gene data frame, where ROIs (regions of interest) are defined by the Power brain parcellation. The Power parcellation defines 264 brain regions, out of which we are able to map AHBA gene expression samples in to 248 ROIs.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(ahba)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">"data.frame"</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ahba)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>]   <span class="dv">248</span> <span class="dv">19227</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>ahba[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>       <span class="dv">7249</span>   <span class="dv">51312</span>   <span class="dv">54708</span>   <span class="dv">56172</span>    <span class="dv">151295</span>  <span class="dv">254910</span>     <span class="dv">7486</span>  <span class="dv">643858</span>     <span class="dv">4296</span>   <span class="dv">55323</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span>   <span class="sc">-</span><span class="fl">0.0559</span>  <span class="fl">0.9041</span>  <span class="fl">0.2480</span>  <span class="fl">0.6225</span> <span class="sc">-</span><span class="fl">0.058305</span> <span class="sc">-</span><span class="fl">0.2234</span> <span class="sc">-</span><span class="fl">0.04350</span> <span class="sc">-</span><span class="fl">0.0687</span> <span class="sc">-</span><span class="fl">0.20332</span>  <span class="fl">0.1353</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="dv">10</span>  <span class="sc">-</span><span class="fl">0.0801</span> <span class="sc">-</span><span class="fl">0.1264</span>  <span class="fl">0.1999</span> <span class="sc">-</span><span class="fl">0.1327</span> <span class="sc">-</span><span class="fl">0.723591</span> <span class="sc">-</span><span class="fl">0.3544</span>  <span class="fl">0.10761</span>  <span class="fl">0.2484</span> <span class="sc">-</span><span class="fl">0.22604</span> <span class="sc">-</span><span class="fl">0.0250</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="dv">100</span>  <span class="fl">0.0766</span>  <span class="fl">0.0693</span> <span class="sc">-</span><span class="fl">0.0651</span>  <span class="fl">0.0465</span> <span class="sc">-</span><span class="fl">0.401925</span> <span class="sc">-</span><span class="fl">0.3309</span> <span class="sc">-</span><span class="fl">0.00106</span>  <span class="fl">0.0383</span> <span class="sc">-</span><span class="fl">0.00497</span> <span class="sc">-</span><span class="fl">0.0148</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="dv">101</span> <span class="sc">-</span><span class="fl">0.1631</span>  <span class="fl">0.1146</span> <span class="sc">-</span><span class="fl">0.0210</span> <span class="sc">-</span><span class="fl">0.2545</span> <span class="sc">-</span><span class="fl">0.000259</span>  <span class="fl">0.0274</span> <span class="sc">-</span><span class="fl">0.16150</span> <span class="sc">-</span><span class="fl">0.1989</span>  <span class="fl">0.57247</span>  <span class="fl">0.0365</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="dv">102</span>  <span class="fl">0.3529</span>  <span class="fl">0.0731</span> <span class="sc">-</span><span class="fl">0.2145</span> <span class="sc">-</span><span class="fl">0.1050</span> <span class="sc">-</span><span class="fl">0.463868</span> <span class="sc">-</span><span class="fl">0.5105</span> <span class="sc">-</span><span class="fl">0.07380</span> <span class="sc">-</span><span class="fl">0.4077</span>  <span class="fl">0.23011</span> <span class="sc">-</span><span class="fl">0.0918</span></span></code></pre></div>
<ul>
<li>COBRE fMRI network data</li>
</ul>
<p>Note that ROI 75 is missing in COBRE data.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>X_cobre <span class="ot">&lt;-</span> COBRE.data<span class="sc">$</span>X.cobre</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>y_cobre <span class="ot">&lt;-</span> COBRE.data<span class="sc">$</span>Y.cobre</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(X_cobre) <span class="ot">&lt;-</span> <span class="fu">getEdgeLabel</span>(<span class="at">node =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">74</span>, <span class="dv">76</span><span class="sc">:</span><span class="dv">264</span>))</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(X_cobre) <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">124</span>)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>y_cobre[y_cobre <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(y_cobre)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>y_cobre</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a> <span class="dv">0</span>  <span class="dv">1</span> </span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="dv">70</span> <span class="dv">54</span> </span></code></pre></div>
<ul>
<li>Gene set collections (GSCs) from MSigDB v7.0</li>
</ul>
<p>Genes are identified by entrez ID.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(gscv7<span class="fl">.0</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="dv">8</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(gscv7<span class="fl">.0</span>)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">"position"</span> <span class="st">"kegg"</span>     <span class="st">"reactome"</span> <span class="st">"tft"</span>      <span class="st">"bp"</span>       <span class="st">"cc"</span>       <span class="st">"mf"</span>       <span class="st">"immune"</span>  </span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(gscv7<span class="fl">.0</span><span class="sc">$</span>kegg)</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="dv">186</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>gscv7<span class="fl">.0</span><span class="sc">$</span>kegg[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="sc">$</span>KEGG_GLYCOLYSIS_GLUCONEOGENESIS</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a> [<span class="dv">1</span>] <span class="st">"55902"</span>  <span class="st">"2645"</span>   <span class="st">"5232"</span>   <span class="st">"5230"</span>   <span class="st">"5162"</span>   <span class="st">"5160"</span>   <span class="st">"5161"</span>   <span class="st">"55276"</span>  <span class="st">"7167"</span>   <span class="st">"84532"</span> </span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>[<span class="dv">11</span>] <span class="st">"2203"</span>   <span class="st">"125"</span>    <span class="st">"3099"</span>   <span class="st">"126"</span>    <span class="st">"3098"</span>   <span class="st">"3101"</span>   <span class="st">"127"</span>    <span class="st">"5224"</span>   <span class="st">"128"</span>    <span class="st">"5223"</span>  </span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>[<span class="dv">21</span>] <span class="st">"124"</span>    <span class="st">"230"</span>    <span class="st">"501"</span>    <span class="st">"92483"</span>  <span class="st">"5313"</span>   <span class="st">"160287"</span> <span class="st">"2023"</span>   <span class="st">"5315"</span>   <span class="st">"5214"</span>   <span class="st">"669"</span>   </span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>[<span class="dv">31</span>] <span class="st">"5106"</span>   <span class="st">"5105"</span>   <span class="st">"219"</span>    <span class="st">"217"</span>    <span class="st">"218"</span>    <span class="st">"10327"</span>  <span class="st">"8789"</span>   <span class="st">"5213"</span>   <span class="st">"5211"</span>   <span class="st">"3948"</span>  </span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>[<span class="dv">41</span>] <span class="st">"2597"</span>   <span class="st">"2027"</span>   <span class="st">"2026"</span>   <span class="st">"441531"</span> <span class="st">"131"</span>    <span class="st">"130"</span>    <span class="st">"3945"</span>   <span class="st">"220"</span>    <span class="st">"221"</span>    <span class="st">"222"</span>   </span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>[<span class="dv">51</span>] <span class="st">"223"</span>    <span class="st">"224"</span>    <span class="st">"130589"</span> <span class="st">"226"</span>    <span class="st">"1738"</span>   <span class="st">"1737"</span>   <span class="st">"229"</span>    <span class="st">"57818"</span>  <span class="st">"3939"</span>   <span class="st">"2538"</span>  </span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>[<span class="dv">61</span>] <span class="st">"5236"</span>   <span class="st">"2821"</span>  </span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a><span class="sc">$</span>KEGG_CITRATE_CYCLE_TCA_CYCLE</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a> [<span class="dv">1</span>] <span class="st">"3420"</span>   <span class="st">"1743"</span>   <span class="st">"5106"</span>   <span class="st">"1431"</span>   <span class="st">"5162"</span>   <span class="st">"5105"</span>   <span class="st">"5160"</span>   <span class="st">"5161"</span>   <span class="st">"283398"</span> <span class="st">"2271"</span>  </span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>[<span class="dv">11</span>] <span class="st">"6392"</span>   <span class="st">"4967"</span>   <span class="st">"6390"</span>   <span class="st">"3419"</span>   <span class="st">"6391"</span>   <span class="st">"3418"</span>   <span class="st">"3417"</span>   <span class="st">"48"</span>     <span class="st">"47"</span>     <span class="st">"4191"</span>  </span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>[<span class="dv">21</span>] <span class="st">"1738"</span>   <span class="st">"4190"</span>   <span class="st">"1737"</span>   <span class="st">"55753"</span>  <span class="st">"5091"</span>   <span class="st">"6389"</span>   <span class="st">"8802"</span>   <span class="st">"8803"</span>   <span class="st">"8801"</span>   <span class="st">"3421"</span>  </span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>[<span class="dv">31</span>] <span class="st">"50"</span>    </span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a><span class="sc">$</span>KEGG_PENTOSE_PHOSPHATE_PATHWAY</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a> [<span class="dv">1</span>] <span class="st">"6120"</span>   <span class="st">"22934"</span>  <span class="st">"55276"</span>  <span class="st">"25796"</span>  <span class="st">"5634"</span>   <span class="st">"8789"</span>   <span class="st">"5213"</span>   <span class="st">"5211"</span>   <span class="st">"6888"</span>   <span class="st">"7086"</span>  </span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>[<span class="dv">11</span>] <span class="st">"2203"</span>   <span class="st">"84076"</span>  <span class="st">"5226"</span>   <span class="st">"64080"</span>  <span class="st">"226"</span>    <span class="st">"230"</span>    <span class="st">"229"</span>    <span class="st">"9563"</span>   <span class="st">"729020"</span> <span class="st">"221823"</span></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>[<span class="dv">21</span>] <span class="st">"5631"</span>   <span class="st">"51071"</span>  <span class="st">"2539"</span>   <span class="st">"5236"</span>   <span class="st">"8277"</span>   <span class="st">"5214"</span>   <span class="st">"2821"</span>  </span></code></pre></div>
<ul>
<li>DiGSeE scores of GSCs for use in the post-hoc evaluation.</li>
</ul>
<p>DiGSeE scores of schizophrenia relevance are constructed for each GSC. GSCs are filtered by size. We keep gene sets with at least 5 genes when deriving DiGSeE scores for the gene sets. Further details on how we obtained these scores are available in the Supplementary methods to the manuscript.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(digsee)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="dv">8</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(digsee)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">"tft"</span>      <span class="st">"kegg"</span>     <span class="st">"bp"</span>       <span class="st">"cc"</span>       <span class="st">"mf"</span>       <span class="st">"reactome"</span> <span class="st">"position"</span> <span class="st">"immune"</span>  </span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(digsee<span class="sc">$</span>kegg)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>                                    prop sumScore avgScore</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>KEGG_GLYCOLYSIS_GLUCONEOGENESIS   <span class="fl">0.0909</span>    <span class="fl">0.965</span>   <span class="fl">0.0877</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>KEGG_CITRATE_CYCLE_TCA_CYCLE      <span class="fl">0.0000</span>    <span class="fl">0.000</span>   <span class="fl">0.0000</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>KEGG_PENTOSE_PHOSPHATE_PATHWAY    <span class="fl">0.2000</span>    <span class="fl">0.953</span>   <span class="fl">0.1907</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>KEGG_GALACTOSE_METABOLISM         <span class="fl">0.0000</span>    <span class="fl">0.000</span>   <span class="fl">0.0000</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>KEGG_FATTY_ACID_METABOLISM        <span class="fl">0.2857</span>    <span class="fl">1.967</span>   <span class="fl">0.2810</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>KEGG_STEROID_HORMONE_BIOSYNTHESIS <span class="fl">0.2500</span>    <span class="fl">2.277</span>   <span class="fl">0.2846</span></span></code></pre></div>
</div>
<div id="brainclass-network-classification" class="section level1">
<h1 class="hasAnchor">
<a href="#brainclass-network-classification" class="anchor"></a><code>brainClass</code> network classification</h1>
<p>For example, let us use the KEGG pathways to construct gene set expression networks.</p>
<div id="obtain-gene-set-edge-groups" class="section level2">
<h2 class="hasAnchor">
<a href="#obtain-gene-set-edge-groups" class="anchor"></a>Obtain gene set edge groups</h2>
<ol style="list-style-type: decimal">
<li>Downsize gene expression data (Optional but recommended)</li>
</ol>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>ahba <span class="ot">&lt;-</span> <span class="fu">filterGeneExpr</span>(ahba)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>ahba <span class="ot">&lt;-</span> ahba[<span class="fu">rownames</span>(ahba)[<span class="fu">order</span>(<span class="fu">as.numeric</span>(<span class="fu">rownames</span>(ahba)))], </span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>    ]</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>ahba <span class="ot">&lt;-</span> ahba[<span class="sc">-</span><span class="fu">which</span>(<span class="fu">rownames</span>(ahba) <span class="sc">==</span> <span class="st">"75"</span>), ]</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ahba)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>]  <span class="dv">247</span> <span class="dv">4807</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Get a gene set collection (GSC) of interest and filter by gene set sizes</li>
</ol>
<p>The filtering on GSC by gene set sizes is optional by recommended.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">filterGeneSets</span>(<span class="at">geneSetList =</span> gscv7<span class="fl">.0</span><span class="sc">$</span>kegg, <span class="at">candidateGenes =</span> <span class="fu">colnames</span>(ahba), </span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">min.size =</span> <span class="dv">5</span>, <span class="at">max.size =</span> <span class="cn">Inf</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">sapply</span>(kegg, length))</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>      <span class="dv">5</span>       <span class="dv">9</span>      <span class="dv">15</span>      <span class="dv">21</span>      <span class="dv">27</span>     <span class="dv">109</span> </span></code></pre></div>
<p>If one wishes to keep all gene sets from the GSC, set <code>min.size</code> to 2 and <code>max.size</code> to Inf. The <code>min.size</code> parameter cannot be smaller than 2 in order to calculate correlations. Note that, it is still necessary to filter out genes that do not have gene expression information in the reference transcriptional data set, which is the (filtered) AHBA dataset.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Not run</span>
<span class="va">kegg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterGeneSets.html">filterGeneSets</a></span><span class="op">(</span>geneSetList <span class="op">=</span> <span class="va">gscv7.0</span><span class="op">$</span><span class="va">kegg</span>, candidateGenes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ahba</span><span class="op">)</span>, 
    min.size <span class="op">=</span> <span class="fl">2</span>, max.size <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Get gene set edge groups</li>
</ol>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">keggEdgeGrp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getGeneSetEdgeGroup.html">getGeneSetEdgeGroup</a></span><span class="op">(</span>geneExpr <span class="op">=</span> <span class="va">ahba</span>, geneSetList <span class="op">=</span> <span class="va">kegg</span>, 
    cutoff <span class="op">=</span> <span class="fl">0.99</span><span class="op">)</span></code></pre></div>
</div>
<div id="network-classification-and-prediction" class="section level2">
<h2 class="hasAnchor">
<a href="#network-classification-and-prediction" class="anchor"></a>Network classification and prediction</h2>
<p>Take one fold from one repeat of 10-fold cross validation as an example:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">edgesToClassify</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="va">keggEdgeGrp</span><span class="op">)</span>
<span class="va">edgesToClassify</span> <span class="op">&lt;-</span> <span class="va">edgesToClassify</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/duplicated.html">duplicated</a></span><span class="op">(</span><span class="va">edgesToClassify</span><span class="op">)</span><span class="op">]</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span>
<span class="va">cvfolds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getCVfolds.html">getCVfolds</a></span><span class="op">(</span><span class="va">y_cobre</span>, k <span class="op">=</span> <span class="fl">10</span>, repeats <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="va">trainid</span> <span class="op">&lt;-</span> <span class="va">cvfolds</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">k</span> <span class="op">!=</span> <span class="fl">1</span><span class="op">)</span>
<span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/brainclass.html">brainclass</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">X_cobre</span><span class="op">[</span><span class="va">trainid</span><span class="op">$</span><span class="va">Ind</span>, <span class="va">edgesToClassify</span><span class="op">]</span>, 
    y <span class="op">=</span> <span class="va">y_cobre</span><span class="op">[</span><span class="va">trainid</span><span class="op">$</span><span class="va">Ind</span><span class="op">]</span>, edgeGrp <span class="op">=</span> <span class="va">keggEdgeGrp</span><span class="op">)</span></code></pre></div>
<p>Prediction on the test set</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>testid <span class="ot">&lt;-</span> cvfolds[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(k <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>test <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit, <span class="at">X =</span> X_cobre[testid<span class="sc">$</span>Ind, edgesToClassify], </span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">"class"</span>)</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(test, y_cobre[testid<span class="sc">$</span>Ind])</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>test <span class="dv">0</span> <span class="dv">1</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>   <span class="dv">0</span> <span class="dv">7</span> <span class="dv">0</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>   <span class="dv">1</span> <span class="dv">0</span> <span class="dv">6</span></span></code></pre></div>
</div>
</div>
<div id="post-hoc-interpretation" class="section level1">
<h1 class="hasAnchor">
<a href="#post-hoc-interpretation" class="anchor"></a>Post-hoc Interpretation</h1>
<div id="example-1" class="section level2">
<h2 class="hasAnchor">
<a href="#example-1" class="anchor"></a>Example 1</h2>
<p>First, as an example, we evaluate the edge selection results by <em>glmnet::glmnet</em> applied on the COBRE data set by differnt metrics with KEGG pathways.</p>
<ol style="list-style-type: decimal">
<li>Edge selection results by <em>glmnet</em>
</li>
</ol>
<p>Obtain the edge labels of selected features, that is, edges with a non-zero fitted coefficient.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># install.package('glmnet')</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmnet)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>cvfit <span class="ot">&lt;-</span> <span class="fu">cv.glmnet</span>(X_cobre, y_cobre, <span class="at">family =</span> <span class="st">"binomial"</span>, <span class="at">nfolds =</span> <span class="dv">10</span>)</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>selectedEdges <span class="ot">&lt;-</span> <span class="fu">coef</span>(cvfit, <span class="at">s =</span> <span class="st">"lambda.min"</span>)</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>selectedEdges <span class="ot">&lt;-</span> <span class="fu">names</span>(selectedEdges[selectedEdges<span class="sc">@</span>i <span class="sc">+</span> <span class="dv">1</span>, ])[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(selectedEdges)</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="dv">86</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Obtain edgewise metrics</li>
</ol>
<p>Evaluate by KEGG pathways:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">metrics</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/posthoc.edge.html">posthoc.edge</a></span><span class="op">(</span>selected.edgeLabels <span class="op">=</span> <span class="va">selectedEdges</span>, 
    all.edgeLabels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">X_cobre</span><span class="op">)</span>, geneSetList <span class="op">=</span> <span class="va">kegg</span>, geneExpr <span class="op">=</span> <span class="va">ahba</span>, 
    iter <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>
<span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">signif</a></span><span class="op">(</span><span class="va">metrics</span>, <span class="fl">2</span><span class="op">)</span>, width <span class="op">=</span> <span class="st">"90%"</span>, options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>scrollX <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-3ff76f1cbebcb849a875" style="width:90%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-3ff76f1cbebcb849a875">{"x":{"filter":"none","data":[["KEGG_GLYCOLYSIS_GLUCONEOGENESIS","KEGG_CITRATE_CYCLE_TCA_CYCLE","KEGG_PENTOSE_PHOSPHATE_PATHWAY","KEGG_GALACTOSE_METABOLISM","KEGG_FATTY_ACID_METABOLISM","KEGG_STEROID_HORMONE_BIOSYNTHESIS","KEGG_OXIDATIVE_PHOSPHORYLATION","KEGG_PURINE_METABOLISM","KEGG_PYRIMIDINE_METABOLISM","KEGG_ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM","KEGG_GLYCINE_SERINE_AND_THREONINE_METABOLISM","KEGG_CYSTEINE_AND_METHIONINE_METABOLISM","KEGG_VALINE_LEUCINE_AND_ISOLEUCINE_DEGRADATION","KEGG_LYSINE_DEGRADATION","KEGG_ARGININE_AND_PROLINE_METABOLISM","KEGG_HISTIDINE_METABOLISM","KEGG_TRYPTOPHAN_METABOLISM","KEGG_GLUTATHIONE_METABOLISM","KEGG_STARCH_AND_SUCROSE_METABOLISM","KEGG_N_GLYCAN_BIOSYNTHESIS","KEGG_O_GLYCAN_BIOSYNTHESIS","KEGG_AMINO_SUGAR_AND_NUCLEOTIDE_SUGAR_METABOLISM","KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE","KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE","KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_HEPARAN_SULFATE","KEGG_GLYCEROLIPID_METABOLISM","KEGG_INOSITOL_PHOSPHATE_METABOLISM","KEGG_GLYCEROPHOSPHOLIPID_METABOLISM","KEGG_ETHER_LIPID_METABOLISM","KEGG_ARACHIDONIC_ACID_METABOLISM","KEGG_LINOLEIC_ACID_METABOLISM","KEGG_ALPHA_LINOLENIC_ACID_METABOLISM","KEGG_SPHINGOLIPID_METABOLISM","KEGG_GLYCOSPHINGOLIPID_BIOSYNTHESIS_LACTO_AND_NEOLACTO_SERIES","KEGG_GLYCOSPHINGOLIPID_BIOSYNTHESIS_GLOBO_SERIES","KEGG_GLYCOSPHINGOLIPID_BIOSYNTHESIS_GANGLIO_SERIES","KEGG_PYRUVATE_METABOLISM","KEGG_PROPANOATE_METABOLISM","KEGG_BUTANOATE_METABOLISM","KEGG_ONE_CARBON_POOL_BY_FOLATE","KEGG_NICOTINATE_AND_NICOTINAMIDE_METABOLISM","KEGG_FOLATE_BIOSYNTHESIS","KEGG_RETINOL_METABOLISM","KEGG_PORPHYRIN_AND_CHLOROPHYLL_METABOLISM","KEGG_NITROGEN_METABOLISM","KEGG_AMINOACYL_TRNA_BIOSYNTHESIS","KEGG_METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450","KEGG_DRUG_METABOLISM_CYTOCHROME_P450","KEGG_BIOSYNTHESIS_OF_UNSATURATED_FATTY_ACIDS","KEGG_ABC_TRANSPORTERS","KEGG_RIBOSOME","KEGG_RNA_DEGRADATION","KEGG_DNA_REPLICATION","KEGG_SPLICEOSOME","KEGG_PPAR_SIGNALING_PATHWAY","KEGG_BASE_EXCISION_REPAIR","KEGG_MAPK_SIGNALING_PATHWAY","KEGG_ERBB_SIGNALING_PATHWAY","KEGG_CALCIUM_SIGNALING_PATHWAY","KEGG_CYTOKINE_CYTOKINE_RECEPTOR_INTERACTION","KEGG_CHEMOKINE_SIGNALING_PATHWAY","KEGG_PHOSPHATIDYLINOSITOL_SIGNALING_SYSTEM","KEGG_NEUROACTIVE_LIGAND_RECEPTOR_INTERACTION","KEGG_CELL_CYCLE","KEGG_OOCYTE_MEIOSIS","KEGG_P53_SIGNALING_PATHWAY","KEGG_UBIQUITIN_MEDIATED_PROTEOLYSIS","KEGG_SNARE_INTERACTIONS_IN_VESICULAR_TRANSPORT","KEGG_LYSOSOME","KEGG_ENDOCYTOSIS","KEGG_PEROXISOME","KEGG_MTOR_SIGNALING_PATHWAY","KEGG_APOPTOSIS","KEGG_CARDIAC_MUSCLE_CONTRACTION","KEGG_VASCULAR_SMOOTH_MUSCLE_CONTRACTION","KEGG_WNT_SIGNALING_PATHWAY","KEGG_DORSO_VENTRAL_AXIS_FORMATION","KEGG_NOTCH_SIGNALING_PATHWAY","KEGG_HEDGEHOG_SIGNALING_PATHWAY","KEGG_TGF_BETA_SIGNALING_PATHWAY","KEGG_AXON_GUIDANCE","KEGG_VEGF_SIGNALING_PATHWAY","KEGG_FOCAL_ADHESION","KEGG_ECM_RECEPTOR_INTERACTION","KEGG_CELL_ADHESION_MOLECULES_CAMS","KEGG_ADHERENS_JUNCTION","KEGG_TIGHT_JUNCTION","KEGG_GAP_JUNCTION","KEGG_COMPLEMENT_AND_COAGULATION_CASCADES","KEGG_ANTIGEN_PROCESSING_AND_PRESENTATION","KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY","KEGG_NOD_LIKE_RECEPTOR_SIGNALING_PATHWAY","KEGG_RIG_I_LIKE_RECEPTOR_SIGNALING_PATHWAY","KEGG_CYTOSOLIC_DNA_SENSING_PATHWAY","KEGG_JAK_STAT_SIGNALING_PATHWAY","KEGG_HEMATOPOIETIC_CELL_LINEAGE","KEGG_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY","KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY","KEGG_B_CELL_RECEPTOR_SIGNALING_PATHWAY","KEGG_FC_EPSILON_RI_SIGNALING_PATHWAY","KEGG_FC_GAMMA_R_MEDIATED_PHAGOCYTOSIS","KEGG_LEUKOCYTE_TRANSENDOTHELIAL_MIGRATION","KEGG_INTESTINAL_IMMUNE_NETWORK_FOR_IGA_PRODUCTION","KEGG_CIRCADIAN_RHYTHM_MAMMAL","KEGG_LONG_TERM_POTENTIATION","KEGG_NEUROTROPHIN_SIGNALING_PATHWAY","KEGG_LONG_TERM_DEPRESSION","KEGG_OLFACTORY_TRANSDUCTION","KEGG_TASTE_TRANSDUCTION","KEGG_REGULATION_OF_ACTIN_CYTOSKELETON","KEGG_INSULIN_SIGNALING_PATHWAY","KEGG_GNRH_SIGNALING_PATHWAY","KEGG_PROGESTERONE_MEDIATED_OOCYTE_MATURATION","KEGG_MELANOGENESIS","KEGG_ADIPOCYTOKINE_SIGNALING_PATHWAY","KEGG_TYPE_II_DIABETES_MELLITUS","KEGG_ALDOSTERONE_REGULATED_SODIUM_REABSORPTION","KEGG_VASOPRESSIN_REGULATED_WATER_REABSORPTION","KEGG_PROXIMAL_TUBULE_BICARBONATE_RECLAMATION","KEGG_ALZHEIMERS_DISEASE","KEGG_PARKINSONS_DISEASE","KEGG_AMYOTROPHIC_LATERAL_SCLEROSIS_ALS","KEGG_HUNTINGTONS_DISEASE","KEGG_PRION_DISEASES","KEGG_VIBRIO_CHOLERAE_INFECTION","KEGG_EPITHELIAL_CELL_SIGNALING_IN_HELICOBACTER_PYLORI_INFECTION","KEGG_PATHOGENIC_ESCHERICHIA_COLI_INFECTION","KEGG_LEISHMANIA_INFECTION","KEGG_PATHWAYS_IN_CANCER","KEGG_COLORECTAL_CANCER","KEGG_RENAL_CELL_CARCINOMA","KEGG_PANCREATIC_CANCER","KEGG_ENDOMETRIAL_CANCER","KEGG_GLIOMA","KEGG_PROSTATE_CANCER","KEGG_THYROID_CANCER","KEGG_BASAL_CELL_CARCINOMA","KEGG_MELANOMA","KEGG_BLADDER_CANCER","KEGG_CHRONIC_MYELOID_LEUKEMIA","KEGG_ACUTE_MYELOID_LEUKEMIA","KEGG_SMALL_CELL_LUNG_CANCER","KEGG_NON_SMALL_CELL_LUNG_CANCER","KEGG_SYSTEMIC_LUPUS_ERYTHEMATOSUS","KEGG_PRIMARY_IMMUNODEFICIENCY","KEGG_HYPERTROPHIC_CARDIOMYOPATHY_HCM","KEGG_ARRHYTHMOGENIC_RIGHT_VENTRICULAR_CARDIOMYOPATHY_ARVC","KEGG_DILATED_CARDIOMYOPATHY","KEGG_VIRAL_MYOCARDITIS"],[0,0.0026,0,0,0,0,0.0026,0.0026,0.0026,0.0026,0.0052,0.0026,0.0026,0,0,0,0,0.0026,0.0026,0,0.0052,0,0.0026,0.0026,0,0,0.01,0,0.0026,0,0,0,0.0026,0,0,0.0026,0.0026,0.0052,0.0026,0.0026,0.0052,0,0.0026,0,0,0.0052,0.0026,0.0026,0.0026,0,0.0026,0,0.0026,0,0,0,0.0078,0.0026,0.0078,0.0026,0.0052,0.0078,0.0026,0.0052,0.0026,0,0.0052,0,0.0026,0.0052,0.0026,0,0.0052,0.0026,0.0026,0.0026,0,0,0.0078,0.0052,0.0052,0.0026,0.0052,0.0052,0,0.0052,0,0.0026,0.0052,0,0,0,0,0.0026,0.0052,0.0052,0.0052,0.0026,0.0052,0.0026,0.0026,0.0052,0.0026,0.0078,0.0052,0.0052,0.0052,0,0,0.0078,0.0026,0.0052,0.0026,0.0052,0.0052,0.0052,0.0026,0,0.0026,0.0026,0,0.0026,0.0052,0.0052,0.0026,0.0026,0.0052,0.0026,0.0052,0.0026,0.0026,0.0026,0,0.0052,0,0,0.0052,0.0026,0,0,0.0026,0,0.0026,0.0026,0.0052,0.0078,0.0078,0.0078,0.0026],[1,9.3e-05,1,1,1,1,5.5e-05,0.00028,0.0044,0.18,1.6e-24,9.6e-09,0.076,1,1,1,1,0.0088,0.0031,1,2.5e-23,1,0.0026,3.2e-07,1,1,1.4e-67,1,0.00049,1,1,1,0.12,1,1,0.023,0.00023,3.7e-27,4.7e-09,0.00071,3.2e-23,1,0.0052,1,1,3.4e-30,0.0006,0.12,0.0011,1,0.00099,1,0.004,1,1,1,4.6e-37,0.0078,1.1e-45,0.00023,4.8e-24,2.4e-41,0.0033,9.6e-23,0.0098,1,9e-26,1,0.0025,2.4e-21,0.017,1,3.5e-27,0.16,0.0011,5.6e-05,1,1,8.3e-44,1.2e-25,2e-25,1.3e-05,5e-22,1.5e-26,1,8.4e-24,1,3.5e-05,2.3e-24,1,1,1,1,0.00097,1e-35,1.4e-20,4.7e-29,0.0027,9.8e-25,0.0018,0.00027,3e-27,9.9e-06,2.7e-37,2.1e-24,4.1e-32,3.3e-25,1,1,3e-47,0.026,4.4e-30,0.00013,5.1e-26,1e-32,3.3e-23,0.00041,1,1.7e-06,6.3e-06,1,0.00039,5.9e-26,5.8e-27,1.6e-05,2.6e-07,2.9e-25,0.0021,5.3e-25,0.018,0.022,9.8e-06,1,3.7e-17,1,1,2.4e-30,5.3e-08,1,1,1.1e-06,1,0.00023,0.00048,5.5e-28,7.3e-48,9.2e-43,3.1e-46,1.1e-06],[7.4,8.5,8,8,7.2,7.1,6,12,8,8.5,7.8,6.6,9.9,7.6,20,15,6.1,8.9,8.7,8.6,8.4,12,7.3,5.9,5.3,9.5,6.2,9.3,7.2,8.4,5.8,9.6,8.3,8,6.1,7.8,7.8,6.9,6,9,6.3,6.7,7.4,7.1,13,6.6,6.3,7.4,8.3,7.2,5.7,8.5,6.2,8.8,13,6,13,5.9,24,7.5,8.4,5.8,6.7,6.8,11,5.6,15,5.8,8.6,9.2,6.5,7.9,6,7.4,7.6,10,7.2,6.6,6.8,7.1,11,5.2,7.5,7.9,6.8,19,11,6,7.1,8.1,5.7,6.1,7.4,5.9,8.9,8,5.7,9.1,9.1,6,8.4,9.9,6.4,7,12,6.6,9,7.7,11,12,8.9,7.4,6.2,5.9,5.3,6,7.7,8,7.8,28,7.5,18,11,4.8,6.9,5.8,6.9,7,9.2,6.7,5.9,6.2,8.2,7.3,8.1,8.7,7.8,6.7,7.7,5.8,9.2,8.1,6.9,7.6,5.8,9.3,9.6,7.8,5.9],[0.01,1.2e-06,0.043,5.2e-07,0.097,0.52,1,2.7e-35,1.9e-11,3.7e-12,0.95,1,9.4e-13,1.9e-06,3.1e-66,1.2e-53,1,2.9e-12,8.7e-30,0.18,1.3e-07,8.8e-44,0.91,1,1,1.4e-16,1,3.1e-15,0.93,0.17,1,2.7e-19,5.7e-12,0.32,1,0.039,0.0058,1,1,3.1e-12,1,1,0.045,0.99,6.2e-33,1,1,0.005,4.4e-37,0.42,1,0.6,1,2.1e-10,2.9e-54,1,5.5e-28,1,2e-80,0.052,0.0044,1,1,1,5e-21,1,1e-25,1,1.8e-20,5.9e-25,1,1.2e-07,1,0.59,0.95,3.2e-13,1,1,1,0.99,6.9e-22,1,0.67,0.01,0.99,6.3e-45,1.6e-31,1,0.91,9.9e-07,1,1,0.99,1,2.1e-06,4.6e-09,1,1.1e-30,1e-20,1,0.0011,7.1e-22,1,1,3.1e-43,1,0.063,0.14,1.6e-07,1.6e-51,1.4e-19,0.8,1,1,1,1,0.2,5.7e-11,0.37,1.3e-91,0.64,2.5e-82,1.2e-24,1,0.99,1,1,0.98,4.6e-22,1,1,1,3.6e-20,0.74,1.7e-07,3.3e-17,2.8e-07,1,0.44,1,6.6e-10,7.7e-09,1,0.021,1,8.6e-17,1.7e-28,0.019,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Jaccard<\/th>\n      <th>Jaccard.PValue<\/th>\n      <th>avg.edge.btw<\/th>\n      <th>edge.btw.PValue<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><ol start="3" style="list-style-type: decimal">
<li>Association with prior knowledge - the DiGSeE database</li>
</ol>
<p>Get “truth” - KEGG pathways with a top 5% in both average and sum DiGSeE scores are assumed to be associated with schizophrenia.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>metrics <span class="ot">&lt;-</span> <span class="fu">merge</span>(metrics, digsee[[<span class="st">"kegg"</span>]], <span class="at">by =</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">truth =</span> ((avgScore <span class="sc">&gt;=</span> <span class="fu">quantile</span>(avgScore, <span class="fl">0.95</span>)) <span class="sc">+</span> </span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>        (sumScore <span class="sc">&gt;=</span> <span class="fu">quantile</span>(sumScore, <span class="fl">0.95</span>)) <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">+</span> <span class="dv">0</span>)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(metrics<span class="sc">$</span>truth)</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="dv">0</span>   <span class="dv">1</span> </span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="dv">137</span>  <span class="dv">12</span> </span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>metrics<span class="sc">$</span>Row.names[metrics<span class="sc">$</span>truth <span class="sc">==</span> <span class="dv">1</span>]</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a> [<span class="dv">1</span>] <span class="st">"KEGG_ALZHEIMERS_DISEASE"</span>                      <span class="st">"KEGG_AMYOTROPHIC_LATERAL_SCLEROSIS_ALS"</span>      </span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a> [<span class="dv">3</span>] <span class="st">"KEGG_ARGININE_AND_PROLINE_METABOLISM"</span>         <span class="st">"KEGG_CALCIUM_SIGNALING_PATHWAY"</span>              </span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a> [<span class="dv">5</span>] <span class="st">"KEGG_HUNTINGTONS_DISEASE"</span>                     <span class="st">"KEGG_LINOLEIC_ACID_METABOLISM"</span>               </span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a> [<span class="dv">7</span>] <span class="st">"KEGG_LONG_TERM_POTENTIATION"</span>                  <span class="st">"KEGG_MAPK_SIGNALING_PATHWAY"</span>                 </span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a> [<span class="dv">9</span>] <span class="st">"KEGG_NEUROACTIVE_LIGAND_RECEPTOR_INTERACTION"</span> <span class="st">"KEGG_PATHWAYS_IN_CANCER"</span>                     </span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>[<span class="dv">11</span>] <span class="st">"KEGG_PRION_DISEASES"</span>                          <span class="st">"KEGG_TRYPTOPHAN_METABOLISM"</span>                  </span></code></pre></div>
<p>Generate the receiver operating characteristic (ROC) curve for each metric:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">rocs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getROC.html">getROC</a></span><span class="op">(</span>truth <span class="op">=</span> <span class="va">metrics</span><span class="op">$</span><span class="va">truth</span>, test.metric <span class="op">=</span> <span class="va">metrics</span><span class="op">$</span><span class="va">Jaccard</span>, 
    step.size <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span>

<span class="fu">ggplot</span><span class="op">(</span><span class="va">rocs</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">fpr</span>, y <span class="op">=</span> <span class="va">tpr</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_abline</span><span class="op">(</span>slope <span class="op">=</span> <span class="fl">1</span>, 
    intercept <span class="op">=</span> <span class="fl">0</span>, color <span class="op">=</span> <span class="st">"darkred"</span>, linetype <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"FPR"</span>, 
    y <span class="op">=</span> <span class="st">"TPR"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="brainClass_files/figure-html/unnamed-chunk-17-1.png" width="672"></p>
<p>Also, calculate the area under this ROC curve (AUC):</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>DescTools<span class="sc">::</span><span class="fu">AUC</span>(rocs<span class="sc">$</span>fpr, rocs<span class="sc">$</span>tpr)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="fl">0.609</span></span></code></pre></div>
<p>Notice that, with p-valued metrics, signigicance is indicated by smaller values. We thereby need the “one-minus” transformation when calcuating the ROC curve:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">rocs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getROC.html">getROC</a></span><span class="op">(</span>truth <span class="op">=</span> <span class="va">metrics</span><span class="op">$</span><span class="va">truth</span>, test.metric <span class="op">=</span> <span class="fl">1</span> <span class="op">-</span> <span class="va">metrics</span><span class="op">$</span><span class="va">Jaccard.PValue</span>, 
    step.size <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span>

<span class="fu">ggplot</span><span class="op">(</span><span class="va">rocs</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">fpr</span>, y <span class="op">=</span> <span class="va">tpr</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_abline</span><span class="op">(</span>slope <span class="op">=</span> <span class="fl">1</span>, 
    intercept <span class="op">=</span> <span class="fl">0</span>, color <span class="op">=</span> <span class="st">"darkred"</span>, linetype <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"FPR"</span>, 
    y <span class="op">=</span> <span class="st">"TPR"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="brainClass_files/figure-html/unnamed-chunk-19-1.png" width="672"></p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>DescTools<span class="sc">::</span><span class="fu">AUC</span>(rocs<span class="sc">$</span>fpr, rocs<span class="sc">$</span>tpr)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="fl">0.509</span></span></code></pre></div>
</div>
<div id="example-2" class="section level2">
<h2 class="hasAnchor">
<a href="#example-2" class="anchor"></a>Example 2</h2>
<p>Next, we evaluate the same selected edges by <em>glmnet</em> with the Jaccard index by different GSCs (say, KEGG, REACTOME, transcription factor targets and positional).</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gscs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"kegg"</span>, <span class="st">"reactome"</span>, <span class="st">"position"</span>, <span class="st">"tft"</span><span class="op">)</span>
<span class="va">rocs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">gscs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">ind</span><span class="op">)</span> <span class="op">{</span>
    <span class="co"># filter the gene sets by size</span>
    <span class="va">gsc_i</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterGeneSets.html">filterGeneSets</a></span><span class="op">(</span>geneSetList <span class="op">=</span> <span class="va">gscv7.0</span><span class="op">[[</span><span class="va">ind</span><span class="op">]</span><span class="op">]</span>, candidateGenes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">ahba</span><span class="op">)</span>, 
        min.size <span class="op">=</span> <span class="fl">5</span>, max.size <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span>
    <span class="co"># obtain Jaccard indices</span>
    <span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/posthoc.edge.html">posthoc.edge</a></span><span class="op">(</span>selected.edgeLabels <span class="op">=</span> <span class="va">selectedEdges</span>, 
        all.edgeLabels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">X_cobre</span><span class="op">)</span>, geneSetList <span class="op">=</span> <span class="va">gsc_i</span>, 
        geneExpr <span class="op">=</span> <span class="va">ahba</span>, get.jaccard <span class="op">=</span> <span class="cn">TRUE</span>, get.betweenness <span class="op">=</span> <span class="cn">FALSE</span>, 
        iter <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>
    <span class="co"># get roc</span>
    <span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html">merge</a></span><span class="op">(</span><span class="va">res</span>, <span class="va">digsee</span><span class="op">[[</span><span class="va">ind</span><span class="op">]</span><span class="op">]</span>, by <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span>
        <span class="fu">mutate</span><span class="op">(</span>truth <span class="op">=</span> <span class="op">(</span><span class="op">(</span><span class="va">avgScore</span> <span class="op">&gt;=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="va">avgScore</span>, <span class="fl">0.95</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
            <span class="op">(</span><span class="va">sumScore</span> <span class="op">&gt;=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="va">sumScore</span>, <span class="fl">0.95</span><span class="op">)</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">+</span> <span class="fl">0</span><span class="op">)</span>
    <span class="va">roc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getROC.html">getROC</a></span><span class="op">(</span>truth <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">truth</span>, test.metric <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">Jaccard</span>, 
        step.size <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span>

    <span class="va">roc</span><span class="op">$</span><span class="va">gsc</span> <span class="op">&lt;-</span> <span class="va">ind</span>
    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">roc</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="va">.</span><span class="op">)</span></code></pre></div>
<p>Visualisation:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggplot</span><span class="op">(</span><span class="va">rocs</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">fpr</span>, y <span class="op">=</span> <span class="va">tpr</span>, color <span class="op">=</span> <span class="va">gsc</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_line</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu">geom_abline</span><span class="op">(</span>slope <span class="op">=</span> <span class="fl">1</span>, intercept <span class="op">=</span> <span class="fl">0</span>, color <span class="op">=</span> <span class="st">"darkred"</span>, 
        linetype <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"FPR"</span>, y <span class="op">=</span> <span class="st">"TPR"</span>, color <span class="op">=</span> <span class="st">"GSC"</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"ROC curves by Jaccard Index"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="brainClass_files/figure-html/unnamed-chunk-21-1.png" width="672"></p>
<p>AUCs:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">group_by</span>(rocs, gsc) <span class="sc">%&gt;%</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">auc =</span> DescTools<span class="sc">::</span><span class="fu">AUC</span>(fpr, tpr))</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="co"># A tibble: 4 x 2</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  gsc        auc</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="sc">&lt;</span>chr<span class="sc">&gt;</span>    <span class="er">&lt;</span>dbl<span class="sc">&gt;</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span> kegg     <span class="fl">0.609</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a><span class="dv">2</span> position <span class="fl">0.609</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a><span class="dv">3</span> reactome <span class="fl">0.637</span></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a><span class="dv">4</span> tft      <span class="fl">0.531</span></span></code></pre></div>
</div>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<p>Aine, C. J., Bockholt, H. J., Bustillo, J. R., Cañive, J. M., Caprihan, A., Gasparovic, C., … &amp; Calhoun, V. D. (2017). Multimodal neuroimaging in schizophrenia: description and dissemination. <em>Neuroinformatics</em>, 15(4), 343-364.</p>
<p>Friedman, J., Hastie, T., &amp; Tibshirani, R. (2010). Regularization paths for generalized linear models via coordinate descent. <em>Journal of statistical software</em>, 33(1), 1.</p>
<p>Hawrylycz, M. J., Lein, E. S., Guillozet-Bongaarts, A. L., Shen, E. H., Ng, L., Miller, J. A., … &amp; Jones, A. R. (2012). An anatomically comprehensive atlas of the adult human brain transcriptome. <em>Nature</em>, 489(7416), 391-399.</p>
<p>Kim, J., So, S., Lee, H. J., Park, J. C., Kim, J. J., &amp; Lee, H. (2013). DigSee: disease gene search engine with evidence sentences (version cancer). <em>Nucleic acids research</em>, 41(W1), W510-W517.</p>
<p>Li, M., Kessler, D., Arroyo, J., Freytag, S., Bahlo, M., Levina, E., &amp; Yang, J. Y. H. (2020). Guiding and interpreting brain network classification with transcriptional data. <em>bioRxiv</em>.</p>
<p>Liberzon, A., Subramanian, A., Pinchback, R., Thorvaldsdóttir, H., Tamayo, P., &amp; Mesirov, J. P. (2011). Molecular signatures database (MSigDB) 3.0. <em>Bioinformatics</em>, 27(12), 1739-1740.</p>
<p>Power, J. D., Cohen, A. L., Nelson, S. M., Wig, G. S., Barnes, K. A., Church, J. A., … &amp; Petersen, S. E. (2011). Functional network organization of the human brain. <em>Neuron</em>, 72(4), 665-678.</p>
<p>Relión, J. D. A., Kessler, D., Levina, E., &amp; Taylor, S. F. (2019). Network classification with applications to brain connectomics. <em>The annals of applied statistics</em>, 13(3), 1648.</p>
<p>Zeng, Y., &amp; Breheny, P. (2016). Overlapping group logistic regression with applications to genetic pathway selection. <em>Cancer informatics</em>, 15, CIN-S40043.</p>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session Information</h1>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>R version <span class="dv">4</span>.<span class="fl">0.5</span> (<span class="dv">2021-03-31</span>)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>Platform<span class="sc">:</span> x86_64<span class="sc">-</span>apple<span class="sc">-</span><span class="fu">darwin17.0</span> (<span class="dv">64</span><span class="sc">-</span>bit)</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>Running under<span class="sc">:</span> macOS Catalina <span class="dv">10</span>.<span class="fl">15.7</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>Matrix products<span class="sc">:</span> default</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>BLAS<span class="sc">:</span>   <span class="er">/</span>System<span class="sc">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>Accelerate.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>Frameworks<span class="sc">/</span>vecLib.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>libBLAS.dylib</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>LAPACK<span class="sc">:</span> <span class="er">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>R.framework<span class="sc">/</span>Versions<span class="sc">/</span><span class="fl">4.0</span><span class="sc">/</span>Resources<span class="sc">/</span>lib<span class="sc">/</span>libRlapack.dylib</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>locale<span class="sc">:</span></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] en_AU.UTF<span class="dv">-8</span><span class="sc">/</span>en_AU.UTF<span class="dv">-8</span><span class="sc">/</span>en_AU.UTF<span class="dv">-8</span><span class="sc">/</span>C<span class="sc">/</span>en_AU.UTF<span class="dv">-8</span><span class="sc">/</span>en_AU.UTF<span class="dv">-8</span></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>attached base packages<span class="sc">:</span></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>other attached packages<span class="sc">:</span></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a> [<span class="dv">1</span>] igraph_1.<span class="fl">2.6</span>      DescTools_0.<span class="fl">99.41</span> glmnet_4<span class="fl">.1</span><span class="dv">-1</span>      Matrix_1<span class="fl">.3</span><span class="dv">-3</span>      graphclass_1<span class="fl">.1</span>   </span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a> [<span class="dv">6</span>] brainClass_0.<span class="fl">1.0</span>  forcats_0.<span class="fl">5.1</span>     stringr_1.<span class="fl">4.0</span>     dplyr_1.<span class="fl">0.6</span>       purrr_0.<span class="fl">3.4</span>      </span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>[<span class="dv">11</span>] readr_1.<span class="fl">4.0</span>       tidyr_1.<span class="fl">1.3</span>       tibble_3.<span class="fl">1.1</span>      ggplot2_3.<span class="fl">3.3</span>     tidyverse_1.<span class="fl">3.1</span>  </span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>[<span class="dv">16</span>] BiocStyle_2.<span class="fl">18.1</span> </span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>loaded via a <span class="fu">namespace</span> (and not attached)<span class="sc">:</span></span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a> [<span class="dv">1</span>] colorspace_2<span class="fl">.0</span><span class="dv">-1</span>    ellipsis_0.<span class="fl">3.2</span>      class_7<span class="fl">.3</span><span class="dv">-19</span>        rprojroot_2.<span class="fl">0.2</span>    </span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a> [<span class="dv">5</span>] fs_1.<span class="fl">5.0</span>            gld_2.<span class="fl">6.2</span>           rstudioapi_0<span class="fl">.13</span>     proxy_0<span class="fl">.4</span><span class="dv">-25</span>       </span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a> [<span class="dv">9</span>] farver_2.<span class="fl">1.0</span>        cvTools_0.<span class="fl">3.2</span>       DT_0<span class="fl">.18</span>             fansi_0.<span class="fl">4.2</span>        </span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>[<span class="dv">13</span>] mvtnorm_1<span class="fl">.1</span><span class="dv">-1</span>       lubridate_1.<span class="fl">7.10</span>    xml2_1.<span class="fl">3.2</span>          codetools_0<span class="fl">.2</span><span class="dv">-18</span>   </span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a>[<span class="dv">17</span>] splines_4.<span class="fl">0.5</span>       cachem_1.<span class="fl">0.4</span>        rootSolve_1.<span class="dv">8</span>.<span class="fl">2.1</span>   robustbase_0<span class="fl">.93</span><span class="dv">-7</span>  </span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>[<span class="dv">21</span>] knitr_1<span class="fl">.33</span>          jsonlite_1.<span class="fl">7.2</span>      broom_0.<span class="fl">7.6</span>         dbplyr_2.<span class="fl">1.1</span>       </span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>[<span class="dv">25</span>] grpregOverlap_2<span class="fl">.2</span><span class="dv">-0</span> BiocManager_1.<span class="fl">30.15</span> compiler_4.<span class="fl">0.5</span>      httr_1.<span class="fl">4.2</span>         </span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>[<span class="dv">29</span>] backports_1.<span class="fl">2.1</span>     assertthat_0.<span class="fl">2.1</span>    fastmap_1.<span class="fl">1.0</span>       cli_2.<span class="fl">5.0</span>          </span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>[<span class="dv">33</span>] formatR_1<span class="fl">.9</span>         htmltools_0.<span class="dv">5</span>.<span class="fl">1.1</span>   tools_4.<span class="fl">0.5</span>         gtable_0.<span class="fl">3.0</span>       </span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a>[<span class="dv">37</span>] glue_1.<span class="fl">4.2</span>          lmom_2<span class="fl">.8</span>            Rcpp_1.<span class="fl">0.6</span>          cellranger_1.<span class="fl">1.0</span>   </span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a>[<span class="dv">41</span>] jquerylib_0.<span class="fl">1.4</span>     pkgdown_1.<span class="fl">6.1</span>       vctrs_0.<span class="fl">3.8</span>         iterators_1.<span class="fl">0.13</span>   </span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a>[<span class="dv">45</span>] crosstalk_1.<span class="fl">1.1</span>     grpreg_3.<span class="fl">3.1</span>        xfun_0<span class="fl">.22</span>           rbibutils_2.<span class="fl">1.1</span>    </span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a>[<span class="dv">49</span>] rvest_1.<span class="fl">0.0</span>         lifecycle_1.<span class="fl">0.0</span>     DEoptimR_1<span class="fl">.0</span><span class="dv">-8</span>      MASS_7<span class="fl">.3</span><span class="dv">-54</span>        </span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a>[<span class="dv">53</span>] scales_1.<span class="fl">1.1</span>        ragg_1.<span class="fl">1.2</span>          hms_1.<span class="fl">0.0</span>           expm_0<span class="fl">.999</span><span class="dv">-6</span>       </span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a>[<span class="dv">57</span>] yaml_2.<span class="fl">2.1</span>          Exact_2<span class="fl">.1</span>           memoise_2.<span class="fl">0.0</span>       sass_0.<span class="fl">4.0</span>         </span>
<span id="cb24-38"><a href="#cb24-38" aria-hidden="true" tabindex="-1"></a>[<span class="dv">61</span>] stringi_1.<span class="fl">6.1</span>       highr_0<span class="fl">.9</span>           desc_1.<span class="fl">3.0</span>          foreach_1.<span class="fl">5.1</span>      </span>
<span id="cb24-39"><a href="#cb24-39" aria-hidden="true" tabindex="-1"></a>[<span class="dv">65</span>] e1071_1<span class="fl">.7</span><span class="dv">-6</span>         boot_1<span class="fl">.3</span><span class="dv">-28</span>         shape_1.<span class="fl">4.5</span>         Rdpack_2.<span class="fl">1.1</span>       </span>
<span id="cb24-40"><a href="#cb24-40" aria-hidden="true" tabindex="-1"></a>[<span class="dv">69</span>] rlang_0.<span class="fl">4.11</span>        pkgconfig_2.<span class="fl">0.3</span>     systemfonts_1.<span class="fl">0.2</span>   evaluate_0<span class="fl">.14</span>      </span>
<span id="cb24-41"><a href="#cb24-41" aria-hidden="true" tabindex="-1"></a>[<span class="dv">73</span>] lattice_0<span class="fl">.20</span><span class="dv">-44</span>     htmlwidgets_1.<span class="fl">5.3</span>   labeling_0.<span class="fl">4.2</span>      tidyselect_1.<span class="fl">1.1</span>   </span>
<span id="cb24-42"><a href="#cb24-42" aria-hidden="true" tabindex="-1"></a>[<span class="dv">77</span>] magrittr_2.<span class="fl">0.1</span>      bookdown_0<span class="fl">.22</span>       R6_2.<span class="fl">5.0</span>            generics_0.<span class="fl">1.0</span>     </span>
<span id="cb24-43"><a href="#cb24-43" aria-hidden="true" tabindex="-1"></a>[<span class="dv">81</span>] DBI_1.<span class="fl">1.1</span>           pillar_1.<span class="fl">6.0</span>        haven_2.<span class="fl">4.1</span>         withr_2.<span class="fl">4.2</span>        </span>
<span id="cb24-44"><a href="#cb24-44" aria-hidden="true" tabindex="-1"></a>[<span class="dv">85</span>] survival_3<span class="fl">.2</span><span class="dv">-11</span>     modelr_0.<span class="fl">1.8</span>        crayon_1.<span class="fl">4.1</span>        utf8_1.<span class="fl">2.1</span>         </span>
<span id="cb24-45"><a href="#cb24-45" aria-hidden="true" tabindex="-1"></a>[<span class="dv">89</span>] rmarkdown_2<span class="fl">.8</span>       grid_4.<span class="fl">0.5</span>          readxl_1.<span class="fl">3.1</span>        data.table_1.<span class="fl">14.0</span>  </span>
<span id="cb24-46"><a href="#cb24-46" aria-hidden="true" tabindex="-1"></a>[<span class="dv">93</span>] reprex_2.<span class="fl">0.0</span>        digest_0.<span class="fl">6.27</span>       textshaping_0.<span class="fl">3.4</span>   munsell_0.<span class="fl">5.0</span>      </span>
<span id="cb24-47"><a href="#cb24-47" aria-hidden="true" tabindex="-1"></a>[<span class="dv">97</span>] bslib_0.<span class="fl">2.5</span>        </span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Mengbo Li.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
